### ISSUE TITLE
Title translation not working after upgrading to React Navigation v.2

### ISSUE THREAD

**Issue by loukandr** [i18next/react-i18next#438]
Below  snippet worked fine with React Navigation v.1. After upgrading to v.2, it stopped translating any message related to screenProps. Translations in render method work fine though.
In addition, if you navigate to another screen and then come back, translation will take place.
That means it is the immediate translation for screenProps that is broken.

```
static navigationOptions = ({ screenProps }) => ({
    title: screenProps.t('screenTitle'),
    headerRight: (
      <HeaderButtons size={21} OverflowIcon={<Icon name="md-more" size={23} />}>
        {!screenProps.i18n.language.startsWith('en') && (
          <HeaderButtons.Item
            title="English"
            show="never"
            onPress={() => {
              screenProps.i18n.changeLanguage('en');
            }}
          />
        )}
        {!screenProps.i18n.language.startsWith('ar') && (
          <HeaderButtons.Item
            title="Arabic"
            show="never"
            onPress={() => {
              screenProps.i18n.changeLanguage('ar');
            }}
          />
        )}
        console.log(screenProps.i18n.language);
      </HeaderButtons>
    )
  });
```

**Comment by jamuhl** [comment]
And what can i do now to monkey patch react-navigation? used react-navigation myself and i know the pain that comes with using it...

Does the navigationOptions function called on initial render of that page transition?

**Comment by loukandr** [comment]
I can understand your frustration but yes, it does.
It's actually implemented same way as in your React Navigation example code.

**Comment by jamuhl** [comment]
i guess it's related to: https://github.com/react-navigation/react-navigation/issues/4271

Just updated the sample in this repo - can reproduce - but got no solution until that issue gets solved.

**Comment by loukandr** [comment]
I can confirm this issue has been resolved with React Navigation v2.3
